app/controllers/settings_controller.rb

```diff
@@ -4,9 +4,7 @@ class SettingsController < ApplicationController
     controller.ensure_logged_in t("layouts.notifications.you_must_log_in_to_view_your_settings")
   end
 
-  before_filter :except => :unsubscribe do |controller|
-    controller.ensure_authorized t("layouts.notifications.you_are_not_authorized_to_view_this_content")
-  end
+  ensure_can_access_person :person_id, except: [:unsubscribe]
```

Comment:
This is confusing that ensure_can_access_person expects a parameter whose value resolves to *username* and it gets passed param name called `:person_id`. With quick search I see this parameter name being referenced in only one other place so why not just rename the route parameter to avoid further confusion?